packages/repository/src/__tests__/unit/repositories/define-repository-class.unit.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Constructor} from '@loopback/context';
+import {expect} from '@loopback/testlab';
+import {
+  AndClause,
+  AnyObject,
+  Condition,
+  Count,
+  CrudRepository,
+  DataObject,
+  DefaultCrudRepository,
+  DefaultKeyValueRepository,
+  defineCrudRepositoryClass,
+  defineEntityCrudRepositoryClass,
+  defineKeyValueRepositoryClass,
+  Entity,
+  Filter,
+  juggler,
+  model,
+  Model,
+  OrClause,
+  property,
+} from '../../..';
+
+describe('RepositoryClass builder', () => {
+  describe('defineCrudRepositoryClass', () => {
+    it('should generate CRUD repository class', async () => {
+      const AddressRepository = defineCrudRepositoryClass(
+        Address,
+        DummyCrudRepository,
+      );
+      expect(AddressRepository.prototype.find).to.be.a.Function();
```

Comment:
Nice, I see that `AddressRepository.prototype` is typed by TypeScript as `CrudRepositoryClass<Address, DummyCrudRepository<any>>.prototype: DummyCrudRepository<any>` and the `find` method has the right type too: `DummyCrudRepository<any>.find(filter?: Filter<any> | undefined, options?: AnyObject | undefined): Promise<any[]>`.

I guess that's a good alternative how to verify that DummyRepository's method can be called.

Do you think the check is enough to verify that custom repository methods (e.g. `DummyCrudRepository#findByTitle`) will be included in `AddressRepository` prototype too? I would prefer to verify this explicitly: Add `findByTitle` method to `DummyCrudRepository`, verify `AddressRepository.prototype.findById` is a function.